### PR TITLE
Resource Quota Manager add list options

### LINKED ISSUE #65378: Resource Quota Manager missed caculating Uninitialized pods
**Is this a BUG REPORT or FEATURE REQUEST?**:

 /kind bug

**What happened**:
Resource Quota Manager missed caculating Uninitialized pods.

**What you expected to happen**:
Resource Quota Manager takes Uninitialized pods into account.
The quota used status should record Uninitialized pods.

**How to reproduce it (as minimally and precisely as possible)**:
### 1. create a resource quota
```
apiVersion: v1
kind: ResourceQuota
metadata:
  name: test-quota
spec:
  hard:
    cpu: "1"
    memory: 500Mi
    pods: "5"
    resourcequotas: "1"
```
### 2. create an uninitialized pod
```
apiVersion: v1
kind: Pod
metadata:
  name: test
  initializers:
    pending:
    - name: podtest.example.com
spec:
  containers:
    - name: redis
      image: redis:3.0.5
      resources:
        requests:
          cpu: "0.5"
          memory: 500Mi
        limits:
          cpu: "0.5"
          memory: 500Mi
```
APIServer will return a 504 error but pod has already created, you can list pods with "IncludeUninitialized=true" option.
### 3. view quota usage
At first, quota usage is as expected.
```
apiVersion: v1
kind: ResourceQuota
metadata:
  name: test-quota
  namespace: default
spec:
  hard:
    cpu: "1"
    memory: 500Mi
    pods: "5"
    resourcequotas: "1"
status:
  hard:
    cpu: "1"
    memory: 500Mi
    pods: "5"
    resourcequotas: "1"
  used:
    cpu: "0.5"
    memory: 500Mi
    pods: "1"
    resourcequotas: "1"
```
Unfortunately, after a while , the quota usage changed as follows.
```
apiVersion: v1
kind: ResourceQuota
metadata:
  name: test-quota
  namespace: default
spec:
  hard:
    cpu: "1"
    memory: 500Mi
    pods: "5"
    resourcequotas: "1"
status:
  hard:
    cpu: "1"
    memory: 500Mi
    pods: "5"
    resourcequotas: "1"
  used:
    cpu: "0"
    memory: 0
    pods: "0"
    resourcequotas: "1"
```

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
 v 1.9.6
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by DylanBLE**
**What this PR does / why we need it**:
Resource Quota Manager should be able to caculate Uninitialized pods.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #65378 

**Special notes for your reviewer**:

**Release note**:

```release-note
None
```

### CODE REVIEW

File: cmd/kube-controller-manager/app/controllermanager.go [hunk 1]
```diff
@@ -227,6 +229,10 @@ type ControllerContext struct {
 	// InformerFactory gives access to informers for the controller.
 	InformerFactory informers.SharedInformerFactory
 
+	// Extend InformerFactory gives access to informers for the controller w
+	// eg: With {IncludeUninitialized: true} to see all unInitialized resour
```
Comment (lavalamp) [inline cmd/kube-controller-manager/app/controllermanager.go]: These comments are cut off, I don't understand them?

File: cmd/kube-controller-manager/app/controllermanager.go [hunk 2]
```diff
@@ -395,6 +401,18 @@ func CreateControllerContext(s *config.CompletedConfig, rootClientBuilder, clien
 	versionedClient := rootClientBuilder.ClientOrDie("shared-informers")
 	sharedInformers := informers.NewSharedInformerFactory(versionedClient, ResyncPeriod(s)())
 
+	// Resource quota controller needs to see all Uninitialized resources.
```
Comment (lavalamp) [inline cmd/kube-controller-manager/app/controllermanager.go]: Aren't we removing this concept? Do we need to implement this? cc @smarterclayton

File: cmd/kube-controller-manager/app/controllermanager.go [hunk 3]
```diff
@@ -227,6 +229,10 @@ type ControllerContext struct {
 	// InformerFactory gives access to informers for the controller.
 	InformerFactory informers.SharedInformerFactory
 
+	// Extend InformerFactory gives access to informers for the controller w
+	// eg: With {IncludeUninitialized: true} to see all unInitialized resour
+	ExtendInformerFactory informers.SharedInformerFactory
```
Comment (lavalamp) [inline cmd/kube-controller-manager/app/controllermanager.go]: If it's for uninitialized resources, call it "IncludeUninitializedInformerFactory". "Extend" doesn't help the reader.

File: cmd/kube-controller-manager/app/controllermanager.go (hunk 2 above)
Comment (smarterclayton) [reply cmd/kube-controller-manager/app/controllermanager.go]: I think at this point if we can reach agreement on namespace initialization (and its needs) not using the generic mechanism or fields, then we probably will remove the generic mechanism and fields.  Let's put a hold on this until we can have that discussion, maybe next api machinery meeting.

Comment (liggitt) [reply cmd/kube-controller-manager/app/controllermanager.go]: Agree.

**Comment by idealhack** [comment]
@DylanBLE Do you have the bandwidth to follow up on this?

**Comment by liggitt** [comment]
given that the generic initializer feature, and the `includeUninitialized` list option are not planned to graduate alpha, I don't think this is worthwhile to add

/close

**Pull request closed without merging**